src/net_processing.cpp

```diff
@@ -2976,6 +3011,57 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         return;
     }
 
+    // Received from a peer demonstrating readiness to announce transactions via reconciliations.
+    // This feature negotiation should happen: between VERSION and VERACK to avoid relay problems
+    // from switching announcement protocols after the connection is up.
+    if (msg_type == NetMsgType::SENDRECON) {
+        if (!m_txreconciliation) {
+            LogPrint(BCLog::NET, "sendrecon from peer=%d ignored according to our config\n", pfrom.GetId());
+            return;
+        }
+
+        if (pfrom.fSuccessfullyConnected) {
```

Comment:
Hand-testing this requirement between 2 peers on regtest, I wonder if this check is correct.

When we receive peer's VERSION, we reply with VERACK L2707, then we send SENDRECON L2742 so isn't the order always to be VERACK-SENDRECON from a receiver viewpoint and trigger a violation ?

Confirmed by the logs I can observe (additional logs of messages reception) are mine: 

```
2022-06-21T16:00:29Z Received a VERACK from peer=0
2022-06-21T16:00:29Z New outbound peer connected: version: 70016, blocks=0, peer=0 (manual)
2022-06-21T16:00:29Z Received a SENDRECON from peer=0
2022-06-21T16:00:29Z sendrecon received after verack from peer=0; disconnecting
```
This check should be covered in `p2p_sendrecon.py`, dunno if I'm missing something.

Reply:
Can you send me the commit hash where you got these line numbers (L2707, L2742)?
I indeed had this bug in some version, but i fixed it a week or two ago.

Reply:
21bbd5e5d

Effectively it's not the head branch. Though it was subtle enough to be missed during some previous round of reviews, myself included :/
I'll do manual tests again if the PR is ready again for review.